### PR TITLE
Improved code readability and efficiency in `gen_python_ref_pages` and `github_scraper` modules

### PR DESCRIPTION
**Pull request by Y-Srivaishnavi**
This PR introduces the following enhancements:

1. Added type hints and return types:
    Enhances code readability and maintainability.
    Improves static type checking capabilities.

2. Optimized return_details function:
    Replaced unnecessary conversion to list with a set data structure. `target_urls` are not being accessed by index after removing duplicates. Code in`github_scrapers`:
      Old code: `target_urls = list(set(target_urls))`
      New Code: `target_urls = set(target_urls)` 
    Reduces memory usage by avoiding redundant data storage.
    By incorporating these changes, we can improve the overall quality and performance of the codebase.

### CODE REVIEW

**Review by PGijsbers — COMMENTED** [review]
Overall looks good, but the `List` and `Dict` annotations are [deprecated](https://docs.python.org/3/library/typing.html#typing.Dict) in favor of `list`, `dict`, and so on. 

We would much appreciate an update with the new-style annotation, if possible! 😄

File: scripts/github_scraper.py [hunk 1]
```diff
@@ -36,11 +37,11 @@ def get_github_info(target_url):
     return name, description, stars
 
 
-def return_details(target_urls):
+def return_details(target_urls:List[str])->Dict[str, Dict[str, str]]:
```
Comment (PGijsbers) [inline scripts/github_scraper.py:40]:
```suggestion
def return_details(target_urls:list[str])->dict[str, dict[str, str]]:
```
Like this, for example.

Comment (Y-Srivaishnavi) [reply scripts/github_scraper.py:40]: Thank you for pointing this out! Will work on it👍

**Comment by Y-Srivaishnavi** [comment]
I've worked on the changes suggested by @PGijsbers. Please review the updated version and let me know what you think😄

**Review by PGijsbers — APPROVED** [review]
LGTM. Thanks! :)

**Pull request merged**